### DISCUSSION TITLE
contextAPI가 한걸음 늦는 이유

### DISCUSSION THREAD

**Discussion by chsua** [woowacourse-teams/2023-votogether#257] [벌레잡기]
현재 로그아웃을 구현하던 중 context에서는 로그인 정보가 변화했지만 layout에서는 그 정보가 동기화되지 않아 로그인/로그아웃 둘다 새로고침을 해야만 변화된 정보가 렌더링되는 문제점이 있습니다! 어떻게 해결하면 좋을까요?

버그과정
- 로그인을 함
\> 토큰이 저장되는 것 확인
\> contextAPI에 유저정보가 담기는것 확인
\> 하지만 렌더링은 로그인을 해주세요 상태

- 로그아웃을 함
\> 토큰이 삭제되는 것 확인
\> contextAPI에 유저정보가 삭제되는것 확인
\> 하지만 렌더링은 로그인 상태

AuthProvider안에서   
```
useEffect(() => {
    console.log('loggedInfo///-----------------------', loggedInfo);
  }, [loggedInfo]);
```
확인했을때 loggedInfo에서 유저 정보가 삭제/생성되었다가 다시 생성/삭제되는 것을 확인했습니다..
왜일까요..? 

+++ 추가
로그아웃이 되어도 로그인 정보가 남아있는 문제는 해결했습니다.
기존 userInfo가 로그인 정보에 다시 저장이 되어서 발생하는 문제였습니다.
다만, 아직 로그인을 하는 경우 왜 유저 정보가 생겼다가 다시 사라지는지, 그리곤 새로고침하면 생기는지 알 수 없습니다..😂

**Comment by Gilpop8663** [comment]
리엑트 쿼리에 캐싱된 유저 정보 때문이 아닐까요?

로그아웃 시 useUserInfo에서 불러와서 캐싱한 데이터를 초기화 해주면 변경이 되지 않을까 싶어요

**Comment by inyeong-kang** [comment]
우스 말대로 cacheTime이 default가 5분이라, 로그인하고 바로 로그아웃한 경우에는 캐싱된 유저 정보가 남아있게 되네요..🤔🤔

**Comment by chsua** [comment]
지금은 또 로그인/로그아웃이 바로 처리가 되네요ㅠ 혹시 다른분께서 useUserInfo 코드를 수정해주셨나요??
